src/Client.ts

```diff
@@ -211,15 +201,19 @@ export default class Client {
   ): Promise<Client> {
     const options = defaultOptions(opts)
     const apiClient = createApiClientFromOptions(options)
-    const keys = await loadOrCreateKeysFromOptions(options, wallet, apiClient)
-    // TODO: Properly bootstrap the keystore and replace `loadOrCreateKeysFromOptions`
-    const keystore = await InMemoryKeystore.create(keys)
+    const keystore = await bootstrapKeystore(options, apiClient, wallet)
+    const publicKeyBundle = new PublicKeyBundle(
+      await keystore.getPublicKeyBundle()
```

Comment:
Is the idea here that the public keybundle never changes and calling into Keystore each time is 1) a waste and 2) an async operation

Reply:
That's right. If we introduce changes where the Client might mutate the publicKeyBundle, we will need to be mindful of that and update this code to either not store it in the client or refresh on mutation.